redwood/src/keys.rs

```diff
@@ -45,7 +38,12 @@ pub(crate) fn secret_key_from_cert<'a>(
     // Pull the encryption keys that are compatible with by the standard policy
     // (e.g. not SHA-1) supported by Sequoia, and not revoked.
     // These filter options should be kept in sync with `Helper::decrypt()`.
-    let keys: Vec<_> = filter_keys!(cert.keys().secret(), policy).collect();
+    let keys: Vec<_> = cert
+        .keys()
+        .secret()
+        .with_policy(policy, None)
+        .for_storage_encryption()
```

Comment:
@nwalfield, Is there a way to use the `for_storage_encryption()` filter without also using `with_policy()`? I couldn't find one, and dropping the storage encryption part means the wrong secret key gets picked for decryption...

Reply:
Good point.  @teythoon just raised the same point [over here](https://gitlab.com/sequoia-pgp/sequoia-octopus-librnp/-/issues/94#note_1598255766).

One approach would be to first try the standard policy but fall back to the weaker [null policy](https://docs.sequoia-pgp.org/sequoia_openpgp/policy/struct.NullPolicy.html) if the standard policy doesn't find a valid self signature.  This is important, because a lot of keys only have self signatures that rely on SHA-1, but it should still be possible to use them to decrypt messages.

Reply:
One clarification via IRC, the use of the standard policy first before falling back to the null policy is to avoid downgrade attacks.